Source/ProjectRimFactory/Storage/UI/ITab_Items.cs

```diff
@@ -47,6 +50,28 @@ public ITab_Items()
 
         protected Building_StorageUnitIOPort SelectedIOPort => SelThing as Building_StorageUnitIOPort;
 
+        private static Dictionary<Thing, List<Pawn>> canBeConsumedby = new Dictionary<Thing, List<Pawn>>();
+
+        private static List<Pawn> pawnCanReach_Touch_Deadly = new List<Pawn>();
+
+        private static List<Pawn> pawnCanReach_Oncell_Deadly = new List<Pawn>();

```

Comment:
pawnCanReach_Oncell_Deadly - the ITab can be kept open while the game is running; this can change even with a non-static cached `List`.  We may want to check that real-time, as it were, if the particular line is being displayed.

Hell, I think it's possible path safety can change even when the game is paused (but the only way I can quickly think of is by deliberately trying to break it)

Reply:
this issue seems logical when i read your comment.

but i just can't break it. with all that I tried so far it works perfectly fine.

Reply:
Fire spreading while the ITab is open and the game is running? That's the scenario I see.

Reply:
As far as happening even while the game is paused, it would need some weird modded items or something.  Maybe a grenade that the pawn carries, and if they drop it (from menu, while paused), it starts fires?

Oh, wait, I have it:

Paused game, one pawn walking over closed cryposleep caskets.
While game is paused, set them to attack the casket, doing enough damage to trigger hostile forces appearing.
Hostile forces (I think?) trigger deadly danger.

With the (very nice) IsVisible(), it might be perfectly okay to check that for every item in the display? Or is that not how it works?

Reply:
i was about to clear it on every cycle when i noticed that those two are written every time on Draw
[see here](https://github.com/zymex22/Project-RimFactory-Revived/pull/365/files#diff-27eb2f66e659101d298928f19824b2b91afd054bcbd2f17f0efa9f0cecd08d02R145).

```C#
pawnCanReach_Touch_Deadly = pawns.Where(p => p.CanReach(SelectedMassStorageUnit, PathEndMode.ClosestTouch, Danger.Deadly)).ToList();
pawnCanReach_Oncell_Deadly = pawns.Where(p => p.CanReach(SelectedMassStorageUnit, PathEndMode.OnCell, Danger.Deadly)).ToList();
```

therefor the issue could only occur in the time frame between the tab refresh starting and it ending.
And that is way to small.

so i think we are good here